### PR TITLE
[dashboard] Show loading indicator when switching git providers to add a new project

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

Show loading indicator when switching git providers to add a new project

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/6654

## How to test

1. Go to https://jx-switch-load.staging.gitpod-dev.com/new
2. Switch between Git providers (hint: click on "change" next to **github.com** or **gitlab.com** near the top)

The switch should show the loading indicator while loading (and not the previous list of repos from the old provider)

## Release Notes

```release-note
[dashboard] Show loading indicator when switching git providers to add a new project
```

## Documentation

/uncc

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by:

Associated issue: *#6654*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
I can't authorize GitHub in this preview env but when switching to GitLab I can see the loading page 🎉

**Comment by laushinka** [comment]
Approving but holding off on the lgtm until @gtsiolis takes a look.

/approve

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6711#issuecomment-969095508" title="Approved">laushinka</a>*

Associated issue: *#6654*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [laushinka]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by laushinka — APPROVED** [review]

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: ebe6527ea1084586472c11272d250f63e177b76b</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6711#pullrequestreview-806978917" title="Approved">laushinka</a>*

Associated issue: *#6654*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [laushinka]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
Oh oops I saw the banner that it still needs my review even after approving, so I did that, turns out it then adds the lgtm label 😅 Sorry if @gtsiolis still wanted to take a look!

**Comment by jankeromnes** [comment]
That's perfect, many thanks for the review @laushinka! 🙏 😁

I also flagged @gtsiolis in case there is an opportunity to improve the design of the loading indicator, but we can also do that as a follow-up / later on. The most important thing was to use the loading indicator appropriately, which fixes the issue. ✅

**Comment by gtsiolis** [comment]
Thanks @jankeromnes @laushinka! 🏓 

FWIW, just tried this in staging and this looks like a great MVC (minimum viable change) iteration. ➿ 
1. DOES. THE. JOB. ✔️ 
2. REUSES. EXISTING. PATTERN. ✔️

**Pull request merged**
